src/west/app/main.py

```diff
@@ -578,10 +620,12 @@ def handle_early_arg_errors(self, early_args):
         # gracefully. This provides more user-friendly output than
         # argparse can do on its own.
 
-        if (early_args.command_name and
-            (early_args.command_name not in self.builtins and
-             (not self.extensions or
-              early_args.command_name not in self.extensions))):
+        if (
+            not early_args.help and
```

Comment:
This fixes an issue with using `-h alias`, I thought about the side-effects of this too, could it break in other places?

Reply:
> I thought about the side-effects of this too, could it break in other places?

In doubt, please make it a separate commit. This will make it easy to revert in the worst case. I just tried and a revert would only lose `west -h alias` which would really be minor.

Reply:
Fixed